omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/AdministrationLinksResource1_8.java

```diff
@@ -0,0 +1,180 @@
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8;
+
+import io.swagger.models.Model;
+import io.swagger.models.ModelImpl;
+import io.swagger.models.properties.ArrayProperty;
+import io.swagger.models.properties.ObjectProperty;
+import io.swagger.models.properties.StringProperty;
+import org.openmrs.module.Extension;
+import org.openmrs.module.web.extension.AdministrationSectionExt;
+import org.openmrs.module.webservices.helper.ModuleFactoryWrapper;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingReadableResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ObjectNotFoundException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.module.webservices.rest.web.v1_0.wrapper.openmrs1_8.ModuleAdministrationLinks1_8;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Resource(name = RestConstants.VERSION_1 + "/administrationlinks", supportedClass = ModuleAdministrationLinks1_8.class,
+		supportedOpenmrsVersions = { "1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*", "2.2.*", "2.3.*",
+				"2.4.*", "2.5.*" })
+public class AdministrationLinksResource1_8 extends BaseDelegatingReadableResource<ModuleAdministrationLinks1_8> {
+
+	private static final String ADMIN_LIST_POINT_ID = "org.openmrs.admin.list";
+
+	private static final String MODULE_TITLE = "moduleTitle";
+
+	private static final String LINKS = "administrationLinks";
+
+	private ModuleFactoryWrapper moduleFactoryWrapper = new ModuleFactoryWrapper();
+
+	public void setModuleFactoryWrapper(ModuleFactoryWrapper moduleFactoryWrapper) {
+		this.moduleFactoryWrapper = moduleFactoryWrapper;
+	}
+
+	@Override
+	public ModuleAdministrationLinks1_8 newDelegate() {
+		return null;
+	}
+
+	@Override
+	public ModuleAdministrationLinks1_8 getByUniqueId(String moduleId) {
+		// Assumes that moduleId is an id of a module that user wants to get admin links for.
+		// AdministrationLinksResource has no id by itself as it's not an OpenMRS data object.
+
+		ModuleAdministrationLinks1_8 administrationLinks = getAdministrationLinksForModule(moduleId);
+		if (administrationLinks == null) {
```

Comment:
Object not found. But object with which id?

Reply:
It means that a given module has no administration links registered. Do you want me to use a different exception then?

Reply:
How does one tell the id of the not found object?

Reply:
It's the id of the module sent in GET request

Reply:
How does one tell that from the exception?

Reply:
Good point :) Should I create a new exception type or just throw `GenericRestException` with a custom message?

Reply:
Can't you just use the same exception?

Reply:
I don't understand :D How can I tell the user the id of the not found object through the `ObjectNotFoundException`? If a user makes a call `GET /webservice.module` and the API returns 404 then user know that `webservice.module` specifically doesn't have any links.

Reply:
Ever heard of an exception message?

Reply:
`ObjectNotFoundException` doesn't support setting a message. Should I add such functionality?

Reply:
I've just added it in https://github.com/openmrs/openmrs-module-webservices.rest/pull/484/commits/114cc1beb9e7852751a88fffff9fba0fdd40864c

Reply:
You still did not specify the id.

Reply:
Right, did in https://github.com/openmrs/openmrs-module-webservices.rest/pull/484/commits/bd6853056b1fb15d4164fdfc138a794bc409bf85